WordPress/Sniffs/VIP/ValidatedSanitizedInputSniff.php

```diff
@@ -45,6 +45,26 @@ public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr )
 	{
 		$tokens = $phpcsFile->getTokens();
 
+		/*
+		 * Quoting 'http://php.net/manual/en/reserved.variables.server.php' :
+		 *
+		 * There is no guarantee that every web server will provide any of these;
+		 * servers may omit some, or provide others not listed here.
+		 *
+		 * That said, a large number of these variables are accounted for in
+		 * the » CGI/1.1 specification, so you should be able to expect those.
+		 * http://www.faqs.org/rfcs/rfc3875.html
+		 *
+		 * @see https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/191
+		 */
+		$server_vars = array(
+			'AUTH_TYPE', 'CONTENT_LENGTH', 'CONTENT_TYPE', 'GATEWAY_INTERFACE',
+			'PATH_INFO', 'PATH_TRANSLATED', 'QUERY_STRING', 'REMOTE_ADDR',
+			'REMOTE_HOST', 'REMOTE_IDENT', 'REMOTE_USER', 'REQUEST_METHOD',
+			'SCRIPT_NAME', 'SERVER_NAME', 'SERVER_PORT', 'SERVER_PROTOCOL',
+			'SERVER_SOFTWARE',
+		);
```

Comment:
> I wonder if these are always available in a WP-CLI context, however.

I'm not sure, but some server vars can always be expected within WordPress thanks to [`wp_fix_server_vars()`](https://developer.wordpress.org/reference/functions/wp_fix_server_vars/). Specifically, it makes sure `SERVER_SOFTWARE` and `REQUEST_URI` are always available, and assumes that `SCRIPT_NAME` and `PHP_SELF` exist.

I think that there are other places where WordPress assumes the existence of different `$_SERVER` vars, however, I've found some of these to give notices when running PHPUnit tests.

I think the point of this sniff is that superglobals are untrusted data that could be modified by anything. So, I kind of favor wontfix'ing this. Even if we could establish that all servers always make x server var available, one can still argue that it is best not to expect it, since something else might be un/intentionally unsetting it. Sure, that probably isn't a good idea, but they are superglobals, and so, like all globals, they are fragile (only more-so).